packages/web-lib/hooks/useBalances.tsx

```diff
@@ -77,38 +99,64 @@ function	useBalances(props?: Types.TUseBalancesReq): Types.TUseBalancesRes {
 				const	token = element.token as string;
 				const	balanceOf = results[rIndex++] as BigNumber;
 				const	decimals = results[rIndex++] as number;
-				const	symbol = results[rIndex++] as string;
-				const	price = results[rIndex++] as BigNumber;
+				let		symbol = results[rIndex++] as string;
+				const	price = format.BN(props?.prices?.[toAddress(token)] || ethers.constants.Zero);
+
+				if (toAddress(token) === ETH_TOKEN_ADDRESS) {
+					symbol = 'ETH';
+				}
 				_data[toAddress(token)] = {
 					decimals: Number(decimals),
 					symbol: symbol,
 					raw: balanceOf,
-					rawPrice: lensContract ? price : ethers.constants.Zero,
+					rawPrice: price,
 					normalized: format.toNormalizedValue(balanceOf, Number(decimals)),
-					normalizedPrice: lensContract ? format.toNormalizedValue(price, 6) : 0,
-					normalizedValue: lensContract ? (format.toNormalizedValue(balanceOf, Number(decimals)) * format.toNormalizedValue(price, 6)) : 0
+					normalizedPrice: format.toNormalizedValue(price, 6),
+					normalizedValue: (format.toNormalizedValue(balanceOf, Number(decimals)) * format.toNormalizedValue(price, 6))
 				};
 			}
 			performBatchedUpdates((): void => {
-				set_data(_data);
+				set_rawData(_data);
 				set_error(undefined);
 				set_status({...defaultStatus, isSuccess: true, isFetched: true});
 			});
-			return (_data);
 		} catch (_error) {
 			performBatchedUpdates((): void => {
-				set_data({});
+				set_rawData(defaultData);
 				set_error(_error as Error);
 				set_status({...defaultStatus, isError: true, isFetched: true});
 			});
-			return ({});
 		}
 	// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [isActive, web3Address, props?.chainID, web3ChainID, provider, networks, props?.key]);
+	}, [isActive, web3Address, props?.chainID, web3ChainID, provider, networks, props?.key, ...effectDependencies]);
 	useEffect((): void => {
 		getBalances();
 	}, [getBalances]);
 
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Once the prices are available, we can update each balance with the price
+	** information.
+	**************************************************************************/
+	const updatePriceInformation = useCallback(async (): Promise<void> => {
+		const _data = {...rawData};
+		for (const entries of Object.entries(rawData)) {
```

Comment:
nit: I'd refactor this like so for readability
```suggestion
		for (const [key, value] of Object.entries(rawData)) {
```

Reply:
Looks great 👍